prow/cmd/branchprotector/README.md

```diff
@@ -0,0 +1,170 @@
+# Branch Protection Documentation
+
+branchprotector configures [github branch protection] according to a specified
+policy.
+
+## Policy configuration
+
+Extend the primary prow [`config.yaml`] document to include a top-level
+`branch-protection` key that looks like the following:
+
+```yaml
+
+branch-protection:
+  orgs:
+    kubernetes:
+      repos:
+        test-infra:
+          # Protect all branches in kubernetes/test-infra
+          protect-by-default: true
+          # Always allow the org's oncall-team to push
+          pushers: ["oncall-team"]
+          # Ensure that the extra-process-followed github status context passes.
+          # In addition, adds any always_run: True prow jobs
```

Comment:
It adds any prow jobs that are required to report.
A job is required to report if it does not have `skip_report` set to `true` and if the job is either an `always_run` job, a `run_if_changed` job, or a `run_after_success` job that is the child of a required job.

Reply:
Done